lib/src/zip_decoder.dart

```diff
@@ -36,13 +36,15 @@ class ZipDecoder {
         ..unixPermissions = unixPermissions;
 
       // see https://github.com/brendan-duncan/archive/issues/21
-      if(Platform.isWindows) {
-        file.isFile = !file.name.endsWith('/');
-      } else {
-        // Should this not be removed and rely only on the above check?
+      // UNIX systems has a creator version of 3 decimal at 1 byte offset
+      if(zfh.versionMadeBy >> 8 == 3) {
```

Comment:
I met too many `undefined`s or `unknown`s when I was looking for the non-unix versions of it. 
It is only known for certain what those will be IF the versionMadeBy is UNIX. The external file attributes are system-dependent. It might mean something completely different for XY system.

With that, it is entirely possible that I am missing something but I am more confident in this solution than checking for unix attribs for this particular case.